Community/Tdarr_Plugin_AV1_HandBreak_Transcode.js

```diff
@@ -0,0 +1,185 @@
+
+// define resolution order from ResolutionSelection from biggest to smallest
+const resolutionOrder = ['8KUHD', '4KUHD', '1080p', '720p', '480p'];
+
+// define the width and height of each resolution from the resolution order
+const resolutionsdimensions = {
+  '8KUHD': '--width 7680 --height 4320',
+  '4KUHD': '--width 3840 --height 2160',
+  '1080p': '--width 1920 --height 1080',
+  '720p': '--width 1280 --height 720',
+  '480p': '--width 640 --height 480',
+};
+
+// eslint-disable-next-line no-unused-vars
+const plugin = (file, librarySettings, inputs) => {
+  // eslint-disable-next-line no-unused-vars
+  const importFresh = require('import-fresh');
+  // eslint-disable-next-line no-unused-vars
+  const library = importFresh('../methods/library.js');
+  // eslint-disable-next-line no-unused-vars
+  const lib = require('../methods/lib')();
+
+  // Get the selected resolution from the 'ResolutionSelection' variable
+  const selectedResolution = inputs.ResolutionSelection;
+
+  getMediaInfo(file);
+  // use mediainfo to match height and width to a resolution on resolutiondimensions
+  let dimensions = resolutionsdimensions[selectedResolution];
+  // if the file is smaller than the selected resolution then use the file resolution
+  if (MediaInfo.videoHeight < dimensions.split(' ')[3] || MediaInfo.videoWidth < dimensions.split(' ')[1]) {
+    dimensions = `--width ${MediaInfo.videoWidth} --height ${MediaInfo.videoHeight}`;
+    // eslint-disable-next-line brace-style
+  }
+
+  // if the file is larger than the selected resolution then use the selected resolution
+  else {
+    // loop through the resolution order
+    for (let i = 0; i < resolutionOrder.length; i += 1) {
+      // if the selected resolution is the same as the current resolution in the loop
+      if (selectedResolution === resolutionOrder[i]) {
+        // break the loop
+        break;
+      }
+    }
+  }
+
+  // read the bitrate of the video stream
+  let videoBitRate = MediaInfo.videoBR;
+
+  // if videoBitrate is over 1000000 devide by 100 to get the bitrate in Kbps
+  if (videoBitRate > 1000000) {
+    videoBitRate /= 100;
+  } else { videoBitRate /= 1000; }
+  // if VideoBitrate is smaller than selected bitrate then use the videoBitrate
+  if (videoBitRate < inputs.BitRate) {
+    // eslint-disable-next-line no-param-reassign
+    inputs.BitRate = videoBitRate;
+    // eslint-disable-next-line brace-style
+  }
+  // if VideoBitrate is larger than selected bitrate then use the selected bitrate
+  else {
+    // eslint-disable-next-line no-self-assign, no-param-reassign
+    inputs.BitRate = inputs.BitRate;
+  }
+
+  response.infoLog += '';
+  // eslint-disable-next-line no-constant-condition
+  if ((true) || file.forceProcessing === true) {
+    // eslint-disable-next-line max-len
+    response.preset = `--encoder svt_av1 -b ${inputs.BitRate} -r 24 -E aac -B 320 -R Auto -6 dpl2 --normalize-mix 1 -f ${inputs.Container} --no-optimize ${dimensions} --crop 0:0:0:0`;
```

Comment:
Why is it cropping for 0? Also, plugin doesn't mention it's going to convert everything to AAC or anything else that it's doing in this command.

Reply:
From what I understand Handbrake tries to auto-crop video and this was the way I could think of to disable that if that is incorrect I will just remove it from the preset